backend/bottari/src/main/java/com/bottari/teambottari/service/TeamMemberService.java

```diff
@@ -54,6 +61,38 @@ public List<ReadTeamMemberStatusResponse> getTeamMemberStatusByTeamBottariId(
                 .toList();
     }
 
+    @Transactional
+    public Long joinTeamBottari(
+            final JoinTeamBottariRequest request,
+            final String ssaid
+    ) {
+        final TeamBottari teamBottari = teamBottariRepository.findByInviteCode(request.inviteCode())
+                .orElseThrow(() -> new BusinessException(
+                        ErrorCode.TEAM_BOTTARI_NOT_FOUND,
+                        "해당하는 초대코드 없음"));
+        final Member member = memberRepository.findBySsaid(ssaid)
+                .orElseThrow(() -> new BusinessException(ErrorCode.MEMBER_NOT_FOUND));
+        validateMemberNotInTeam(teamBottari.getId(), ssaid);
+        final TeamMember teamMember = new TeamMember(teamBottari, member);
+        teamMemberRepository.save(teamMember);
+        addTeamMemberSharedItems(teamBottari.getId(), teamMember);
+
+        return teamMember.getId();
+    }
+
+    private void addTeamMemberSharedItems(
+            final Long teamBottariId,
+            final TeamMember teamMember
+    ) {
+        final List<TeamSharedItemInfo> sharedItemInfos = teamSharedItemInfoRepository.findAllByTeamBottariId(
+                teamBottariId
+        );
+        for (final TeamSharedItemInfo info : sharedItemInfos) {
+            final TeamSharedItem teamSharedItem = new TeamSharedItem(info, teamMember);
+            teamSharedItemRepository.save(teamSharedItem);
+        }
```

Comment:
```suggestion
        final List<TeamSharedItem> teamSharedItems = sharedItemInfos.stream()
                .map(info -> new TeamSharedItem(info, teamMember))
                .toList();
        teamSharedItemRepository.saveAll(teamSharedItems);
```

돌면서 저장하기 vs 만들어두고 때려버리기

별론가

물론 identity 전략이라 batch insert는 안되지만?